Sources/Ollama/Tools.swift

```diff
@@ -0,0 +1,126 @@
+import Foundation
+
+extension Chat {
+    
+    /// A tool provided to the LLM
+    public struct Tool {
+        /// The definition included in the API calls
+        let definition: ToolDefinition
+        /// The action to be run when this tool is called on by the AI
+        let action: ([String: Value]) -> Value
+        
+        /// A parameter passed into the function
+        public struct ToolParameter {
```

Comment:
You haven't specified a public initializer, which unfortunately appears to make this feature impossible to be used.

Perhaps a test should be added for initialization that does not use the `@testable` import, to ensure all additions to the API can be properly used by consumers.

Reply:
Thanks, I overlooked that as I did all my testing in the unit tests.